Removes network cards and printers from tablets

JohnFulpWillard opened the pull request:
## About The Pull Request

### Network

There was a lot of unused code around Network cards (because networks as a whole is underused) so a lot of code was gutted here, I tried updating as much as I can to use better code, or to just be carried over at all, but sadly networks are networks.

I also renamed NETWORK_CARDS to NETWORK_TABLETS

It never F'ing worked
![image](https://user-images.githubusercontent.com/53777086/192108315-86a72698-bad4-4342-9e35-35c7186f5573.png)

(This is supposed to say the hardware name instead of SYSTEM, but they never had a hardware name SET, EVER)

Post-my PR
![image](https://user-images.githubusercontent.com/53777086/192108384-0f6353ba-7d0d-4240-a5f7-3448837c6a66.png)

### Printers

- Removes printer parts and affiliated shitcode
- Simplifies emptying paper bins into tablets, adds a new proc to remove papers from paper bins while taking into account already generated paper
- Moves Messenger's photo uploading to the new try_insert proc in hopes to simplify the attackby because I added two new interactions to it for adding papers to tablets, which made it look worse than it currently is.
- Moves Welder and Wrench interactions to welder_act/wrench_act for the same reason

## Why It's Good For The Game

I'm gonna try to keep the ball rolling to remove all PC parts, looking at tablet code makes me cry, and makes me wonder why the hackmd to transition PDA to tablet considered this to be 'good'.

## Changelog

:cl:
balance: You can no longer upgrade your tablets to have signals to the NTnet despite not being anywhere near telecomms. Instead, the Mining base has a lesser connection, and all computers have Ethernet connections.
balance: All tablets now have a printer in them, rather than it being an external part.
removal: NTNet tablet cards has been removed.
/:cl:

JohnFulpWillard committed:
Removes network cards

There was a lot of unused code around Network cards (because networks as a whole is underused) so a lot of code was gutted here, I tried updating as much as I can to use better code, or to just be carried over at all, but sadly networks are networks.

I also renamed NETWORK_CARDS to NETWORK_TABLETS

TheVekter commented:
So is this going to affect how PDAs or tablets work at all?

JohnFulpWillard commented:
> So is this going to affect how PDAs or tablets work at all?

Download speeds will be fast on computers, regular on tablets, slow for tablets on lavaland, won't be able to upgrade your tablet to download faster.
I think that's the only thing that affects players

JohnFulpWillard commented:
Actually I'll include printers in here too to get 2 for 1 like my other Prs

JohnFulpWillard committed:
Removes printer tablet parts

Removes printer parts and affiliated shitcode
Simplifies emptying paper bins into tablets, adds a new proc to remove papers from paper bins while taking into account already generated paper
Moves Messenger's photo uploading to the new try_insert proc in hopes to simplify the attackby
Moves Welder and Wrench interactions to welder_act/wrench_act for the same reason

TheVekter commented:
> Actually I'll include printers in here too to get 2 for 1 like my other Prs

Does killing printers do anything to the Cargo tablets? People still use the label printer to sell shit.

JohnFulpWillard commented:
> Does killing printers do anything to the Cargo tablets? People still use the label printer to sell shit.

All tablets will store paper in the tablets themselves rather than need an external printer

JohnFulpWillard committed:
prettierx

JohnFulpWillard committed:
removes printers from maps

Mothblocks commented on code/modules/modular_computers/computers/item/computer.dm:
```diff
@@ -574,18 +591,28 @@ GLOBAL_LIST_EMPTY(TabletMessengers) // a list of all active messengers, similar
 
+	// physical computers are always connected through ethernet
+	if(istype(physical, /obj/machinery/modular_computer))
+		return 3
```
Can you make these defines?

Mothblocks commented on code/modules/modular_computers/computers/item/tablet.dm:
```diff
@@ -199,6 +199,23 @@
+		return FALSE
+	// no AIs/pAIs
+	var/mob/living/silicon/robot/borgs_ONLY = borgo
+	if(!istype(borgs_ONLY))
```
Is the variable named "borgo" not guaranteed to be a borg? Stupid names (including yours)

JohnFulpWillard replied:
It used to be, it isn't anymore as AIs/pAIs were given this when we moved pda to tablet and they needed a messenger app. var name was me being pissed off about it not being borgs only

Mothblocks replied:
then var name should be renamed to silicon and this variable renamed to borg, and if you dont care enough then just change this to borg_real or something like that but this name is silly

JohnFulpWillard committed:
defines for signals

VladinXXV commented:
How does this affect laptops? I really enjoyed using a laptop over a tablet for the increased hardware capabilities. Printers? Glad to see them go. Network cards? Not so much. The fun of modular computers is the modularity. While I agree some parts are a little pointless, I'd still like to be able to modify computers. I want to feel like the laptop is a fair sidegrade to a tablet. Right to repair and stuff. Also I still want something done about #67338 PLEASE.

JohnFulpWillard committed:
party rockers in the hou

Changes the borgo var on integrated tablets

JohnFulpWillard committed:
Uses hardware flags instead of istype

Laptops now have ethernet connections while on-station, uses hardware_flag instead of checking if the physical is a computer

JohnFulpWillard commented:
> The fun of modular computers is the modularity.

https://github.com/tgstation/tgstation/pull/70012#issuecomment-1255479411
The modularity of tablets will be within the applications, not the parts. All parts are currently planned for removal in one way or another, because parts are too confusing to expect players to fully understand. I've criticized them a lot in the past and my opinion hasn't changed on them,

Your issue technically is not a bug as the messenger app is not supposed to be moved to disks and stuff, but I'm planning on removing individual HDD/SSD and instead having it all be on the tablets themselves.

VladinXXV commented:
> > The fun of modular computers is the modularity.
> 
> [#70012 (comment)](https://github.com/tgstation/tgstation/pull/70012#issuecomment-1255479411) The modularity of tablets will be within the applications, not the parts. All parts are currently planned for removal in one way or another, because parts are too confusing to expect players to fully understand. I've criticized them a lot in the past and my opinion hasn't changed on them,
> 
> Your issue technically is not a bug as the messenger app is not supposed to be moved to disks and stuff, but I'm planning on removing individual HDD/SSD and instead having it all be on the tablets themselves.

What does this mean for laptops as a sidegrade? Their whole thing was more advanced parts at the cost of being bigger. Are they still going to have "hardware" better than a tablet but less than that of a console?

JohnFulpWillard commented:
> What does this mean for laptops as a sidegrade? Their whole thing was more advanced parts at the cost of being bigger. Are they still going to have "hardware" better than a tablet but less than that of a console?

As far as I'm aware, laptops are meant to just be a computer with some tablet stuff like cells and portability. I don't have a clear answer for you now, though.
